projects/plugins/jetpack/_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -1013,6 +1052,72 @@ public static function get_products( $request ) {
 		}
 	}
 
+	/**
+	 * Gets the users licenses counts.
+	 *
+	 * @since 10.4.0
+	 *
+	 * @return string|WP_Error A JSON object of user license counts if the request was successful, or a WP_Error otherwise.
+	 */
+	public static function get_user_license_counts() {
+		$wpcom_request = Client::wpcom_json_api_request_as_user(
+			'/jetpack-licensing/user/licenses/counts',
+			'2',
+			array(
+				'method'  => 'GET',
+				'headers' => array(
+					'Content-Type'    => 'application/json',
+					'X-Forwarded-For' => Jetpack::current_user_ip( true ),
+				),
+			)
+		);
+
+		$response_code = wp_remote_retrieve_response_code( $wpcom_request );
+		if ( 200 === $response_code ) {
+			$license_counts = json_decode( wp_remote_retrieve_body( $wpcom_request ) );
+			return $license_counts;
+		} else {
+			return new WP_Error(
+				'failed_to_fetch_data',
+				esc_html__( 'Unable to fetch the requested data.', 'jetpack' ),
+				array( 'status' => $response_code )
+			);
+		}
+	}
+
+	/**
+	 * Update the user-licenses activation notice dismissal data.
+	 *
+	 * @since 10.4.0
+	 *
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 *
+	 * @return array|WP_Error
+	 */
+	public static function update_licensing_activation_notice_dismiss( $request ) {
+
+		if ( ! isset( $request['last_detached_count'] ) ) {
+			return new WP_Error( 'invalid_param', esc_html__( 'Missing parameter "last_detached_count".', 'jetpack' ), array( 'status' => 404 ) );
+		}
+
+		$default             = array(
+			'last_detached_count' => null,
+			'last_dismissed_time' => null,
+		);
+		$last_detached_count = ( '' === $request['last_detached_count'] )
+			? $default['last_detached_count']
+			: $request['last_detached_count'];
+		// Use UTC timezone and convert to ISO8601 format(DateTime::W3C) for best compatibility with JavaScript Date in all browsers.
+		$last_dismissed_time = ( new DateTime( 'NOW', new DateTimeZone( 'UTC' ) ) )->format( DateTime::W3C );
```

Comment:
Ahh yes, totally makes sense, thank you!
Ok, I created a new permission check function, `user_licensing_permission_check()`, that checks that the user is the connection owner and the user is connected. What do you think? Do you foresee any issues there?

Reply:
> since the time difference calculation is being done client-side in the browser

Ah yes, that's a good point. I suppose if we were to  switch to `wp_timezone()` here, we would then have to switch to using `@wordpress/date` on the client side as well to keep things in sync.

Your approach is probably simpler though :)

Reply:
> Ah yes, that's a good point. I suppose if we were to switch to wp_timezone() here, we would then have to switch to using @wordpress/date on the client side as well to keep things in sync.

Ohh, Does `@wordpress/date` somehow know and use the site's set timezone? I see in the docs it says. "Defaults to timezone from site.".  Ohh, I didn't know that, before, TIL. Although I still don't understand offhand how the @wordpress/date package would know the site's timezone.. 🤔  

Well anyway, yes if you don't mind, I'll just keep it as it is, for simplicity sake. 🙂

Reply:
It's some of the magic of the WordPress packages :). If you type in `wp.date.__experimentalGetSettings().timezone.string` in your browser console while in the block editor or in the Jetpack Dashboard, you should see it in action.

The package lives here:
https://github.com/WordPress/gutenberg/tree/1b0ff7e8606cde2b71270f7744897eb2819eecd7/packages/date

> Although I still don't understand offhand how the @wordpress/date package would know the site's timezone.. 🤔

The info gets passed down from PHP to JS via `wp_localize_script()`, here:
https://github.com/WordPress/WordPress/blob/aa62c253e0639924ad9453ee88d3b9851ccdff4d/wp-includes/script-loader.php#L349-L353

We use the package in some places in the Jetpack dashboard, but not directly to fetch the timezone so far. Here is an example:
https://github.com/Automattic/jetpack/blob/e13487d0dbbe4c743739b6f1a3e37c143a8c2614/projects/plugins/jetpack/_inc/client/components/product-expiration/index.jsx#L53-L60

> I'll just keep it as it is, for simplicity sake. 🙂

👍